src/components/search/search.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react';
+import useSWRInfinite from 'swr/infinite';
+
+import {
+    Button,
+    TextInput,
+    Select,
+    ESystemIconNames,
+    TypographyScale,
+} from '@mdb/flora';
+
+import { ContentPiece } from '../../interfaces/content-piece';
+
+import Results from './results';
+import {
+    titleStyles,
+    searchBoxStyles,
+    sortBoxStyles,
+    loadMoreStyles,
+} from './styles';
+import { SearchProps, SortByType } from './types';
+import { fetcher, sortByOptions } from './utils';
+
+const Search: React.FunctionComponent<SearchProps> = ({
+    name,
+    hideSortBy = false,
+    filters = [],
+}) => {
+    const [search, setSearch] = useState('');
+    const [sortBy, setSortBy] = useState<SortByType>('recent');
+
+    const getKey = (pageIndex: number, previousPageData: ContentPiece[]) => {
+        if (previousPageData && !previousPageData.length) return null;
+        return `?search=${search}&sort=${sortBy}&page=${pageIndex}&filters=${filters.join(
+            ','
+        )}`;
+    };
+    const { data, size, setSize, error, isValidating } = useSWRInfinite(
+        getKey,
+        fetcher,
+        {
+            revalidateIfStale: false,
+            revalidateOnFocus: false,
+            revalidateOnReconnect: false,
+            shouldRetryOnError: false,
+        }
+    );
+
+    return (
+        <>
+            <TypographyScale variant="heading5" sx={titleStyles}>
+                All {name} Content
+            </TypographyScale>
+            <div sx={searchBoxStyles}>
```

Comment:
You could leverage the form tag with a role of search so that we follow some accessibility best practices like illustrated here: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/search#search_form_labels_and_accessibility